### PR TITLE
Constants for status codes, lookup for reason phrases.

### PR DESCRIPTION
**Pull request by Tratcher**
#174

### CODE REVIEW

**Comment by Eilon** [comment]
Looks lovely :shipit:

File: src/Microsoft.AspNet.WebUtilities/StatusCodes.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        public const int Status405MethodNotAllowed = 405;
+        public const int Status406NotAcceptable = 406;
+        public const int Status407ProxyAuthenticationRequired = 407;
+        public const int Status408RequestTimeout = 408;
+        public const int Status409Conflict = 409;
+        public const int Status410Gone = 410;
+        public const int Status411LengthRequired = 411;
+        public const int Status412PreconditionFailed = 412;
+        public const int Status413RequestEntityTooLarge = 413;
+        public const int Status414RequestUurTooLong = 414;
```
Comment (PinpointTownes) [inline src/Microsoft.AspNet.WebUtilities/StatusCodes.cs]: `Uur`?

File: src/Microsoft.AspNet.WebUtilities/StatusCodes.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        public const int Status415UnsupportedMediaType = 415;
+        public const int Status416RequestedRangeNotSatisfiable = 416;
+        public const int Status417ExpectationFailed = 417;
+        public const int Status418ImATeapot = 418;
```
Comment (PinpointTownes) [inline src/Microsoft.AspNet.WebUtilities/StatusCodes.cs]: I love that one, but is it really useful? :smile:

**Comment by davidfowl** [comment]
:shipit:

**Comment by PinpointTownes** [comment]
My 2 cents: the `[Status] + code + name` pattern is a bit hard to read and I'm not sure it's really useful: if you know the status code, you don't need to rely on a constant to use it. And if you wanna use a constant, that's often because you don't want to play with the status code directly.

**Comment by davidfowl** [comment]
@PinpointTownes has a point, should me model it after the crazy System.Net enum?

**Comment by Tratcher** [comment]
The enum naming assumed you (and everyone reading your code) were deeply familiar with the reason phrases and what the resulting status codes would be. If you just use reason phrases people have to guess the matching status code, where if you just use integers they have to guess the matching reason phrase.

I picked the `Status404NotFound` pattern because it lets you see both parts together for validation. It also lets you use intellisense like a lookup table where you can type in either the status code or the reason phrase and it immediately shows you the other half. It has `Status` on there because you can't start with digits.

Would you prefer `NotFound404`? I find that a little odd because it's the opposite of how it shows up in the response and spec. How about `S404NotFound`?

**Comment by Eilon** [comment]
I like it just fine the way it is. I view the point of these as being to eliminate what *appear* to be magic numbers from the code. And I think this does that beautifully.

**Comment by Eilon** [comment]
And for the record, I *strongly* dislike the System.Net StatusCode enum because it adds no value at all. It just takes one hard to understand piece of information and hides it behind another hard to understand piece of information.

**Pull request merged**